torch/nn/modules/module.py

```diff
@@ -1509,78 +1527,105 @@ def _call_impl(self, *args, **kwargs):
+
+            result = forward_call(*args, **kwargs)
+            if _global_forward_hooks or self._forward_hooks:
+                for hook_id, hook in (
+                    *_global_forward_hooks.items(),
+                    *self._forward_hooks.items(),
+                ):
+                    if hook_id in self._forward_hooks_with_kwargs:
+                        hook_result = hook(self, args, kwargs, result)
+                    else:
+                        hook_result = hook(self, args, result)
+
+                    # mark that forced hook has been run
+                    if hook_id in self._forward_hooks_forced:
+                        self._forward_hooks_forced[hook_id] = True
+                    elif hook_id in _global_forward_hooks_forced:
+                        _global_forward_hooks_forced[hook_id] = True
+
+                    if hook_result is not None:
+                        result = hook_result
+
+            if bw_hook:
+                if not isinstance(result, (torch.Tensor, tuple)):
+                    warnings.warn("For backward hooks to be called,"
+                                  " module output should be a Tensor or a tuple of Tensors"
+                                  f" but received {type(result)}")
+                result = bw_hook.setup_output_hook(result)
+
+            # Handle the non-full backward hooks
+            if non_full_backward_hooks:
+                var = result
+                while not isinstance(var, torch.Tensor):
+                    if isinstance(var, dict):
+                        var = next((v for v in var.values() if isinstance(v, torch.Tensor)))
+                    else:
+                        var = var[0]
+                grad_fn = var.grad_fn
+                if grad_fn is not None:
+                    for hook in non_full_backward_hooks:
+                        grad_fn.register_hook(_WrappedHook(hook, self))
+                    self._maybe_warn_non_full_backward_hook(args, result, grad_fn)
+
+            return result
+
+        finally:
+            # run forced hooks if they have not already been run
+            # For now only forward hooks have the force option but perhaps this
+            # functionality should be added to full backward hooks as well.
+            if not all(_global_forward_hooks_forced.values()):
+                for hook_id, hook in _global_forward_hooks.items():
+                    if hook_id in _global_forward_hooks_forced and not _global_forward_hooks_forced[hook_id]:
+                        hook_result = hook(self, args, result)
+
+            if not all(self._forward_hooks_forced.values()):
+                for hook_id, hook in self._forward_hooks.items():
+                    if hook_id in self._forward_hooks_forced and not self._forward_hooks_forced[hook_id]:
+                        if hook_id in self._forward_hooks_with_kwargs:
+                            hook_result = hook(self, args, kwargs, result)
```

Comment:
To confirm, the result of the hook should not be returned here since we aren't actually handling the exception raised so we don't actually care about the result of the hook. Does that sound right?